force-app/main/default/classes/PlannedSalesTriggerHelper.cls

```diff
@@ -0,0 +1,138 @@
+                                               FROM Planned_Sales__c WHERE Id IN:Trigger.New];
+        
+        //Map<Integer,Integer> monthAndYearOfPlannedSale=new Map<Integer,Integer>();
+
+        Set<Integer> month=new Set<Integer>();
+        Set<Integer> year=new Set<Integer>();
+        
+        
+        
+        Map<String,Planned_Sales__c> relatedPlannedSales=new Map<String,Planned_Sales__c>();
+        
+        Map<Id,Planned_Sales__c> plannedSalesToUpdate=new Map<Id,Planned_Sales__c>();
+                
+                
+        if(!createdPlannedSales.isEmpty()){
+            for(Planned_Sales__c plannedSale:createdPlannedSales){
+                String key=plannedSale.Month__c+'_'+plannedSale.Year__c+'_'+plannedSale.Car_Type__c;
+                relatedPlannedSales.put(key,plannedSale);
+                //! Problem HERE
+                month.add((Integer) plannedSale.Month__c);
+                year.add((Integer) plannedSale.Year__c);
+                //monthAndYearOfPlannedSale.put((Integer) plannedSale.Month__c,(Integer) plannedSale.Year__c);
+            }
+
+            if(month!=null && year!=null){
+                for(Opportunity opportunity:[SELECT Id,CloseDate,StageName,Amount,(SELECT Id,Product2.Condition__c FROM OpportunityLineItems) FROM Opportunity 
+                                              WHERE CALENDAR_MONTH(CloseDate) IN :month 
+                                             AND CALENDAR_YEAR(CloseDate) IN:year
+                                             AND StageName='Closed Won'
+                                            ]){ 
+                     
+                    String key=opportunity.CloseDate.month()+'_'+opportunity.CloseDate.year(); 
+                                                 
+                    for(OpportunityLineItem oli:opportunity.OpportunityLineItems){
+                        
+                        String key2=key+'_'+oli.Product2.Condition__c;
+                        
+                        if(relatedPlannedSales.containsKey(key2)){
+                            
+                            Planned_Sales__c plannedSale=relatedPlannedSales.get(key2);
+                            
+                            if(plannedSale!=null){
+                                plannedSale.Total_Amount_Earned__c+=opportunity.Amount;
+                                plannedSale.Count_of_Cars_Sold__c+=opportunity.OpportunityLineItems.size();
+                                
+                                plannedSalesToUpdate.put(plannedSale.Id,plannedSale);
+                            }
+                        }                             
+                    }
+                } 
+            }
+ 
+        }
+        if(!plannedSalesToUpdate.isEmpty()){
+            update plannedSalesToUpdate.values();
+        }
+
+    }
+    public static void updatePlannedSales(List<Planned_Sales__c> newPlannedSales,Map<Id,Planned_Sales__c> oldPlannedSalesMap){
```

Comment:
This method literally do the same an the one on insert, just combine them instead of duplicating the code